src/pages/permissions/CreateIdentityModal.tsx

```diff
@@ -0,0 +1,98 @@
+import { FC, useState } from "react";
+import { ActionButton, Input, Modal } from "@canonical/react-components";
+import { useFormik } from "formik";
+import Loader from "components/Loader";
+import { fetchIdentityDetails } from "api/auth-identities";
+import { base64EncodeTokenDetails } from "util/helpers";
+
+interface Props {
+  onClose: () => void;
+}
+
+const CreateIdentityModal: FC<Props> = ({ onClose }) => {
+  const [isLoadingToken, setIsLoadingToken] = useState(false);
```

Comment:
maybe this state can simply be `isLoading` as it is the only loading state for the modal